Add `remove-node` recipe (#24)
unifiedjs/unifiedjs.github.io: doc/learn/remove-node.md

---
group: recipe
title: Remove a node
description: How to remove a node in any unified flavored syntax tree.
tags:
  - node
  - tree
  - traverse
  - walk

>   - remove

Reviewer: ```suggestion
  - remove
  - delete
```

author: John Letey
published: 2020-03-04
modified: 2020-03-04
---

## How to remove a node

### Contents

*   [What is a node](#what-is-a-node)
*   [Removing a node](#removing-a-node)

### What is a node

A node is a single language specific unit inside a [syntax tree][syntax-tree].

> For example: a heading in markdown, or anchor element in HTML. In unified, nodes

Reviewer: ```suggestion
For example: a heading in markdown, or anchor element in HTML.  In unified, nodes
```

follow the [unist][] specification.

### Removing a node

The concept of removing a node involves
[tree traversal][tree-traversal] of a [syntax tree][syntax-tree].

Unified compatible utilities should be used for finding a node.

> Utilities are functions that work with nodes. All specifications

Reviewer: ```suggestion
Utilities are functions that work with nodes.  All specifications
```

that extend [unist][] can use the [unist utilities][unist-utils],
but they can also have their own utilities for more specific nodes.

> To start removing nodes for your input you'll need:

Reviewer: ```suggestion
To start removing nodes for your input you’ll need:
```

*   A processor (such as [`remark`][remark]).
*   A utility of choice.

For this example we use [`remark`][remark]

> and [`unist-util-remove`][unist-util-remove]. We want to remove

Reviewer: ```suggestion
and [`unist-util-remove`][unist-util-remove].  We want to remove
```
